batch/batch/worker.py

```diff
@@ -846,6 +846,8 @@ def __init__(self):
     async def post_job_complete(self, job, run_duration):
         try:
             await self.post_job_complete_1(job, run_duration)
+        except Exception:
+            log.exception(f'error while posting {job} complete', stack_info=True)
```

Comment:
This isn't handling the error, it's just logging a message. What kinds of exceptions can happen in post_job_complete_1? What do we need to do?

Reply:
I can't possibly know all exceptions that can happen. The retry the write to GCS should help. I can put the GCS write call in a while loop and retry on all Exceptions that are not asyncio Cancels. We already retry posting the status to the worker. The message I was seeing was this one:

```
Task exception was never retrieved\nfuture: <Task finished coro=<Worker.post_job_complete() done, defined at /usr/local/lib/python3.6/site-packages/batch/worker.py:846> exception=ConnectionError(ProtocolError('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer')),)>
```